### ISSUE TITLE
Add new extension definitions to objects

### ISSUE THREAD

**Issue by himynamesdave** [signalscorps/file2stix#57]
https://github.com/signalscorps/file2stix/blob/beta-1/docs/extractions-analysis-mode.md#13-cryptocurrency-observables-indicator

https://github.com/signalscorps/file2stix/blob/beta-1/docs/extractions-analysis-mode.md#14-cve-observables-vulnerability

https://github.com/signalscorps/file2stix/blob/beta-1/docs/extractions-analysis-mode.md#16-credit-card-observables-indicator

https://github.com/signalscorps/file2stix/blob/beta-1/docs/extractions-analysis-mode.md#17-iban-observables-indicator

https://github.com/signalscorps/file2stix/blob/beta-1/docs/extractions-analysis-mode.md#19-sigma-rule-observables-indicator

https://github.com/signalscorps/file2stix/blob/beta-1/docs/extractions-analysis-mode.md#20-cpe-observables-software

These extension definitions are all in https://github.com/signalscorps/stix2-objects

**Comment by himynamesdave** [comment]
@shanthanu9 yes, 

e.g.

https://github.com/signalscorps/file2stix/blob/beta-1/docs/extractions-analysis-mode.md#stix-21-credit-card-sco

```json
{
  "type": "credit-card",
  "spec_version": "2.1",
  "id": "credit-card--<GENERATED BY STIX2 LIBRARY>",
  "created_by_ref": "identity--<IF TLP RED/AMBER/GREEN USER IDENTITY ID>",
  "type": "<CREDIT CARD TYPE>",
  "number": "<CREDIT CARD NUMBER>",
  "extensions": {
    "extension-definition--abd6fc0e-749e-4e6c-a20c-1faa419f5ee4" : {
      "extension_type" : "new-sco"
    }
  },
  "object_marking_refs": [
      "marking-definition--<TLP LEVEL SET>"
  ]
}
```

where extension definition is imported from 

https://github.com/signalscorps/stix2-objects/tree/main/extension-definition/extension-definition--abd6fc0e-749e-4e6c-a20c-1faa419f5ee4

**Comment by himynamesdave** [comment]
@shanthanu9 the actual extension definition (imported from stix2-objects) must be printed in the bundle.

for example the bundle below creates a cryptocurrency sdo with `extension-definition--532ae28d-137b-4b89-afb7-9cf9b504191b`.

However, the actual extension defintion object  `xtension-definition--532ae28d-137b-4b89-afb7-9cf9b504191b` is not in the bundle;

```json
{
    "type": "bundle",
    "id": "bundle--669373f3-dbfe-4945-835b-c126d8a681ad",
    "objects": [
        {
            "type": "identity",
            "spec_version": "2.1",
            "id": "identity--acf55024-6bbe-486f-a27a-7967559324f4",
            "created_by_ref": "identity--a99a2297-3044-4011-9a7e-2ff15e056b65",
            "created": "2022-01-01T00:00:00.000Z",
            "modified": "2022-01-01T00:00:00.000Z",
            "name": "file2stix",
            "description": "https://github.com/signalscorps/file2stix/",
            "identity_class": "organization",
            "sectors": [
                "technology"
            ],
            "contact_information": "https://www.signalscorps.com/contact/",
            "object_marking_refs": [
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
            ]
        },
        {
            "type": "marking-definition",
            "spec_version": "2.1",
            "id": "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9",
            "created": "2017-01-20T00:00:00.000Z",
            "definition_type": "tlp",
            "name": "TLP:WHITE",
            "definition": {
                "tlp": "white"
            }
        },
        {
            "type": "indicator",
            "spec_version": "2.1",
            "id": "indicator--a797eb5f-3a69-465c-b6c5-2a3e6cd75b27",
            "created": "2022-09-21T19:50:49.387299Z",
            "modified": "2022-09-21T19:50:49.387299Z",
            "name": "BTC: 3FZbgi29cpjq2GjdwV8eyHuJJnkLtktZc5",
            "indicator_types": [
                "unknown"
            ],
            "pattern": "[ cryptocurrency:symbol = 'BTC' AND cryptocurrency:address = '3FZbgi29cpjq2GjdwV8eyHuJJnkLtktZc5' ]",
            "pattern_type": "stix",
            "pattern_version": "2.1",
            "valid_from": "2022-09-21T19:50:49.387299Z",
            "external_references": [
                {
                    "source_name": "file2stix",
                    "description": "This object was created using file2stix from the Signals Corps.",
                    "url": "https://github.com/signalscorps/file2stix"
                }
            ],
            "object_marking_refs": [
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
            ]
        },
        {
            "type": "cryptocurrency",
            "spec_version": "2.1",
            "id": "cryptocurrency--2ea13f77-10bc-5b86-90fd-61c7c980eb86",
            "symbol": "BTC",
            "address": "3FZbgi29cpjq2GjdwV8eyHuJJnkLtktZc5",
            "object_marking_refs": [
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
            ],
            "extensions": {
                "extension-definition--532ae28d-137b-4b89-afb7-9cf9b504191b": {
                    "extension_type": "new-sco"
                }
            }
        },
        {
            "type": "observed-data",
            "spec_version": "2.1",
            "id": "observed-data--3bb0758c-81f6-4b4f-949e-35212394cc7f",
            "created_by_ref": "identity--acf55024-6bbe-486f-a27a-7967559324f4",
            "created": "2022-09-21T19:50:49.24947Z",
            "modified": "2022-09-21T19:50:49.24947Z",
            "first_observed": "2022-09-21T19:50:49.24947Z",
            "last_observed": "2022-09-21T19:50:49.24947Z",
            "number_observed": 1,
            "object_refs": [
                "cryptocurrency--2ea13f77-10bc-5b86-90fd-61c7c980eb86"
            ],
            "external_references": [
                {
                    "source_name": "file2stix",
                    "description": "This object was created using file2stix from the Signals Corps.",
                    "url": "https://github.com/signalscorps/file2stix"
                }
            ],
            "object_marking_refs": [
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
            ]
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--14ebe5cb-c89c-41e9-b885-a463d2044d06",
            "created_by_ref": "identity--acf55024-6bbe-486f-a27a-7967559324f4",
            "created": "2022-09-21T19:50:49.24947Z",
            "modified": "2022-09-21T19:50:49.24947Z",
            "relationship_type": "default-extract-from",
            "source_ref": "indicator--a797eb5f-3a69-465c-b6c5-2a3e6cd75b27",
            "target_ref": "report--9de9524f-1c8a-4b72-a616-bd525e5b6fdd",
            "external_references": [
                {
                    "source_name": "file2stix",
                    "description": "This object was created using file2stix from the Signals Corps.",
                    "url": "https://github.com/signalscorps/file2stix"
                }
            ],
            "object_marking_refs": [
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
            ]
        },
        {
            "type": "relationship",
            "spec_version": "2.1",
            "id": "relationship--cd613426-c81e-4f32-9c2c-fc6b3fbf6023",
            "created_by_ref": "identity--acf55024-6bbe-486f-a27a-7967559324f4",
            "created": "2022-09-21T19:50:49.24947Z",
            "modified": "2022-09-21T19:50:49.24947Z",
            "relationship_type": "pattern-contains",
            "source_ref": "indicator--a797eb5f-3a69-465c-b6c5-2a3e6cd75b27",
            "target_ref": "cryptocurrency--2ea13f77-10bc-5b86-90fd-61c7c980eb86",
            "external_references": [
                {
                    "source_name": "file2stix",
                    "description": "This object was created using file2stix from the Signals Corps.",
                    "url": "https://github.com/signalscorps/file2stix"
                }
            ],
            "object_marking_refs": [
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
            ]
        },
        {
            "type": "extension-definition",
            "spec_version": "2.1",
            "id": "extension-definition--c8ea5ecb-f4a3-45e7-94de-9b9ba05161af",
            "created_by_ref": "identity--a99a2297-3044-4011-9a7e-2ff15e056b65",
            "created": "2022-01-01T00:00:00.000Z",
            "modified": "2022-01-01T00:00:00.000Z",
            "name": "MISP Warning Lists",
            "description": "This schema adds MISP Warning List matches to extracted Objects",
            "schema": "https://raw.githubusercontent.com/signalscorps/stix2-objects/main/schemas/properties/misp-warning-list-match/schema.json",
            "version": "1.0",
            "extension_types": [
                "property-extension"
            ],
            "object_marking_refs": [
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
            ]
        },
        {
            "type": "report",
            "spec_version": "2.1",
            "id": "report--9de9524f-1c8a-4b72-a616-bd525e5b6fdd",
            "created_by_ref": "identity--acf55024-6bbe-486f-a27a-7967559324f4",
            "created": "2022-09-21T19:50:49.24947Z",
            "modified": "2022-09-21T19:50:49.24947Z",
            "name": "File converted: crypto_btc.txt",
            "report_types": [
                "threat_report"
            ],
            "published": "2022-09-21T20:50:49.249466Z",
            "object_refs": [
                "identity--acf55024-6bbe-486f-a27a-7967559324f4",
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9",
                "indicator--a797eb5f-3a69-465c-b6c5-2a3e6cd75b27",
                "cryptocurrency--2ea13f77-10bc-5b86-90fd-61c7c980eb86",
                "observed-data--3bb0758c-81f6-4b4f-949e-35212394cc7f",
                "relationship--14ebe5cb-c89c-41e9-b885-a463d2044d06",
                "relationship--cd613426-c81e-4f32-9c2c-fc6b3fbf6023",
                "extension-definition--c8ea5ecb-f4a3-45e7-94de-9b9ba05161af"
            ],
            "external_references": [
                {
                    "source_name": "file2stix",
                    "description": "This object was created using file2stix from the Signals Corps.",
                    "url": "https://github.com/signalscorps/file2stix"
                }
            ],
            "object_marking_refs": [
                "marking-definition--613f2e26-407d-48c7-9eca-b8e91df99dc9"
            ]
        }
    ]
}
```

**Comment by himynamesdave** [comment]
breaking into #82 #81 #83 #84
